File: lib/utils/app_logger.dart
import 'package:flutter/foundation.dart';

class AppLogger {
  static const _prefix = 'PhoneLockdown';

  static void d(String tag, String msg) => debugPrint('[$_prefix/$tag] $msg');
  static void w(String tag, String msg) =>
      debugPrint('[$_prefix/$tag] WARNING: $msg');
  static void e(String tag, String msg, [Object? error]) {
    debugPrint(
        '[$_prefix/$tag] ERROR: $msg${error != null ? ' ($error)' : ''}');
  }
}

File: android/app/src/main/kotlin/app/phonelockdown/AppLogger.kt
package app.phonelockdown

import android.util.Log

object AppLogger {
    private const val PREFIX = "PhoneLockdown"

    fun d(tag: String, msg: String) = Log.d("$PREFIX/$tag", msg)
    fun i(tag: String, msg: String) = Log.i("$PREFIX/$tag", msg)
    fun w(tag: String, msg: String) = Log.w("$PREFIX/$tag", msg)
    fun w(tag: String, msg: String, t: Throwable) = Log.w("$PREFIX/$tag", msg, t)
    fun e(tag: String, msg: String) = Log.e("$PREFIX/$tag", msg)
    fun e(tag: String, msg: String, t: Throwable) = Log.e("$PREFIX/$tag", msg, t)
}
